pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableInstances.java

```diff
@@ -116,16 +117,25 @@ public String getTableInstances(
   }
 
   @GET
-  @Path("/tables/{tableNameWithType}/livebrokers")
+  @Path("/tables/{tableName}/livebrokers")
   @Produces(MediaType.APPLICATION_JSON)
   @ApiOperation(value = "List the brokers serving a table", notes = "List live brokers of the given table based on EV")
   @ApiResponses(value = {
       @ApiResponse(code = 200, message = "Success"),
       @ApiResponse(code = 404, message = "Table not found"),
-      @ApiResponse(code = 500, message = "Internal server error")})
+      @ApiResponse(code = 500, message = "Internal server error")
+  })
   public List<String> getLiveBrokersForTable(
-      @ApiParam(value = "Table name with type", required = true)
-      @PathParam("tableNameWithType") String tableNameWithType) {
-    return _pinotHelixResourceManager.getLiveBrokersForTable(tableNameWithType);
+      @ApiParam(value = "Table name (with or without type)", required = true)
+      @PathParam("tableName") String tableName) {
+    try {
+      return _pinotHelixResourceManager.getLiveBrokersForTable(tableName);
+    } catch (TableNotFoundException tableNotFoundException) {
+      throw new WebApplicationException(String.format("Table=%s not found", tableName), 404);
+    }
+  }
+
+  public void setPinotHelixResourceManager(PinotHelixResourceManager pinotHelixResourceManager) {
```

Comment:
Suggest removing this function and `PinotTableInstancesTest`. The test in `PinotHelixResourceManagerTest` is good enough, no need to mock a `PinotHelixResourceManager` for one line of code